src/gsy_e/models/strategy/market_agents/one_sided_engine.py

```diff
@@ -157,22 +157,24 @@ def event_offer_traded(self, *, trade):
             try:
                 if ConstSettings.MASettings.MARKET_TYPE == SpotMarketTypeEnum.ONE_SIDED.value:
                     # One sided market should subtract the fees
-                    trade_offer_rate = trade.trade_rate - \
-                                       trade.fee_price / trade.traded_energy
+                    trade_offer_rate = trade.trade_rate - trade.fee_price / trade.traded_energy
                 else:
                     # trade_offer_rate not used in two sided markets, trade_bid_info used instead
                     trade_offer_rate = None
-                updated_trade_bid_info = \
-                    self.markets.source.fee_class.update_forwarded_offer_trade_original_info(
-                        trade.offer_bid_trade_info, offer_info.source_offer)
+                    trade.offer_bid_trade_info.original_offer_rate = (
+                        self.markets.source.fee_class.adapt_offer_fees_on_offer_trade(
```

Comment:
Isn't `adapt_offer_fees_on_offer_trade` returning a tuple? It does not return a trade_bid_info any more

Reply:
Indeed it's returning a tuple, I'm unpacking the returned values at the end of the line:
```python
                            trade.offer_bid_trade_info, offer_info.source_offer)[0])
```

Reply:
Hmm, IMO this does not look nice. Unfortunately, I think the refactoring did not improve the readability. Just to be sure: It did not fix anything, it is only because of https://github.com/gridsingularity/gsy-e/pull/1390/files#r813677517 right?

Reply:
@hannesdiedrich yes, the comment you mentioned is the reason for this refactoring. However, I have just had another look and I think that we can further simplify the interface to only return two elements, disregarding the trade_rate